middle_end/flambda/types/structures/row_like.rec.ml

```diff
@@ -47,6 +47,14 @@ struct
   type index =
     | Known of Index.t (** Known x represents the singleton set: { x } *)
     | At_least of Index.t (** At_least x represents the set { y | x \subset y } *)
+    | Range of { at_least : Index.t; at_most : Index.t; }
+    (** Range { at_least; at_most represents the set
+        { x | at_least \subset x /\ x \subset at_most }
+        or { x | at_least \subset x } if at_most is None
```

Comment:
How can an index be `None`, from what I recall, Indexes can be integers at least in some cases...
Also, it's a bit weird to have this redundancy in how to represent `at_least` (i.e. either directly with `At_least`, or with `Range` using a `None` `at_most`), but no way to represent just the `at_most` case.

Reply:
Oups I didn't update the comment. The case was
```OCaml
| Range of { at_least : Index.t; at_most : Index.t option; }
```